api/src/main/scala/dx/api/DxInstanceType.scala

```diff
@@ -480,28 +481,40 @@ object InstanceTypeDB extends DefaultJsonProtocol {
   // view the user account. The compiler may not have these
   // permissions, causing this method to throw an exception.
   // TODO: move this to DxOrg and DxUser objects
-  private def getPricingModel(dxApi: DxApi, billTo: String, region: String): Map[String, Float] = {
+  private def getPricingModel(dxApi: DxApi,
+                              billTo: String,
+                              region: String): Option[Map[String, Float]] = {
     val request = Map("fields" -> JsObject("pricingModelsByRegion" -> JsTrue))
-    val response = billTo match {
-      case _ if billTo.startsWith("org") =>
-        dxApi.orgDescribe(billTo, request)
-      case _ if billTo.startsWith("user") =>
-        dxApi.userDescribe(billTo, request)
-      case _ =>
-        throw new Exception(s"Invalid billTo ${billTo}")
-    }
-    val pricingModelsByRegion = JsUtils.getFields(response, Some("pricingModelsByRegion"))
-    val computeRatesPerHour =
-      JsUtils.getFields(pricingModelsByRegion(region), Some("computeRatesPerHour"))
-    // convert from JsValue to a Map
-    computeRatesPerHour.map {
-      case (name, jsValue) =>
-        val hourlyRate: Float = jsValue match {
-          case JsNumber(x) => x.toFloat
-          case JsString(x) => x.toFloat
-          case _           => throw new Exception(s"compute rate is not a number ${jsValue.prettyPrint}")
-        }
-        name -> hourlyRate
+    Try {
+      billTo match {
+        case _ if billTo.startsWith("org") =>
+          dxApi.orgDescribe(billTo, request)
+        case _ if billTo.startsWith("user") =>
+          dxApi.userDescribe(billTo, request)
+        case _ =>
+          throw new Exception(s"Invalid billTo ${billTo}")
+      }
+    } match {
+      case Success(response) =>
+        JsUtils
+          .getOptionalFields(response, "pricingModelsByRegion")
+          .flatMap(_.get(region))
+          .flatMap(JsUtils.getOptionalFields(_, "computeRatesPerHour"))
+          .filter(_.nonEmpty)
+          .map { computeRatesPerHour =>
+            computeRatesPerHour.map {
+              case (name, jsValue) =>
+                val hourlyRate: Float = jsValue match {
+                  case JsNumber(x) => x.toFloat
+                  case JsString(x) => x.toFloat
+                  case _ =>
+                    throw new Exception(s"compute rate is not a number ${jsValue.prettyPrint}")
+                }
+                name -> hourlyRate
+            }
+          }
+      case Failure(_: dx.PermissionDeniedException) => None
```

Comment:
This would return None on PermissionDeniedException and Option[Map] where the map is Map.empty if permissions ok but the fields we wanted in the response were empty?

Reply:
`Option` has two subclasses, `Some` and `None`. In the case where the response is empty it returns `None` (that is what the `filter` method does`). I also changed it to log other errors and return `None` since the pricing model isn't necessary for the program to function correctly.